File: Aug14/Carl/Q8.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_1s = 0
        counter = 0
        for i in range(0, len(nums)):
            if nums[i] == 1:
                counter += 1
            else:
                if counter > max_1s:
                    max_1s = counter
                counter = 0
        if counter > max_1s:
            max_1s = counter
        return max_1s

s = Solution()
print(s.findMaxConsecutiveOnes([1,1,0,0,1,1,1]))

File: 2018/Aug14/Troy/Q8.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int len = nums.length;
        int result = 0;
        int temp = 0;
        for (int i = 0; i < len; i++) {
            if (nums[i] == 0) {
                if (temp > result) result = temp;
                temp = 0;
            } else {
                temp++;
            }
        }
        return Math.max(temp, result);
    }
}
